src/cdk/datetime/datepicker-input.ts

```diff
@@ -237,7 +245,7 @@ export class CdkDatepickerInput<D> implements AfterContentInit, ControlValueAcce
   constructor(
       protected _elementRef: ElementRef,
       @Optional() public _dateAdapter: DateAdapter<D>,
-      @Optional() @Inject(CDK_DATE_FORMATS) private _dateFormats: CdkDateFormats) {
+      @Optional() @Inject(CDK_DATE_FORMATS) protected _dateFormats: CdkDateFormats) {
```

Comment:
Grab me offline to discuss `CDK_DATE_FORMATS` and `MAT_DATE_FORMATS`

Reply:
Going to separate `CDK_DATE_FORMATS` entirely from `MAT_DATE_FORMATS`.